### PR TITLE
Bugfix/set gas extra field for zero cost tx

### PR DESCRIPTION
**Pull request by liayoo**
- Background: The 'gas' extra field was only set when gas cost > 0, which made transactions with 0 gas price bypass bandwidth limits. 
- Updated to set gas field even when gas cost = 0
- Updated broken tests
- Fixes https://github.com/ainblockchain/ain-blockchain/issues/398

### CODE REVIEW

**Review by minsulee2 — APPROVED** [review]
LGTM!

**Review by platfowner — APPROVED** [review]
LGTM with some minor comments.
Thanks!

File: integration/afan_dapp.test.js [hunk 1]
```diff
@@ -53,8 +53,32 @@ function startServer(application, serverName, envVars, stdioInherit = false) {
...
+          value: 1
+        }
+      ]
+    }
+  }).body.toString('utf-8')).result;
+  assert.deepEqual(ChainUtil.isFailedTx(_.get(createAppRes, 'result')), false);
+  if (!(await waitUntilTxFinalized(serverList, createAppRes.tx_hash))) {
+    console.log(`setUp(): Failed to check finalization of create app tx.`)
+  }
+  const legacySetupRes = parseOrLog(syncRequest('POST', server1 + '/set', {
```
Comment (platfowner) [inline integration/afan_dapp.test.js:81]: The name seems to hint some future work?
If so, what about adding a todo?

Comment (liayoo) [reply integration/afan_dapp.test.js:82]: Done, thanks!

**Comment by liayoo** [comment]
Thanks for the reviews! Now merging..

**Pull request merged**